### PR TITLE
Implemented functionality for inputting into developer and publisher tables

### PR DESCRIPTION
**Pull request by hhm970**
- Wrote more unit tests (planning on writing more if time allows for it).
- Can now insert `developer_id`, `publisher_id` into `game` table entries.

### CODE REVIEW

**Comment by hhm970** [comment]
Closes #129

File: pipeline/load/load.py [hunk 1]
```diff
@@ -30,22 +30,113 @@ def get_db_connection(config) -> connection:
...
+    cursor.execute("""SELECT game_id FROM game
+                        WHERE name = %s""", (inputted_game[0],))
+
+    game_id = cursor.fetchone()['game_id']
+
+    return game_id
+
+
+def get_dev_id(input_game: list, cursor: connection.cursor) -> RealDictRow:
+    """Given a game, returns the corresponding developer_id entry if its developer
```
Comment (shindym) [inline pipeline/load/load.py:54]: Good description

File: pipeline/load/load.py [hunk 2]
```diff
@@ -30,22 +30,113 @@ def get_db_connection(config) -> connection:
...
+        developer_id_match = get_dev_id(input_game, cur)
+
+        if developer_id_match is None:
+
+            cur.execute("""INSERT INTO developer (developer_name)
+                    VALUES (%s)""", (input_game[3],))
+
+            developer_id_match = get_dev_id(input_game, cur)
+
+        cur.close()
```
Comment (shindym) [inline pipeline/load/load.py:81]: Don't believe this needs closing

File: pipeline/load/load.py [hunk 3]
```diff
@@ -30,22 +30,113 @@ def get_db_connection(config) -> connection:
...
+        publisher_id_match = get_pub_id(input_game, cur)
+
+        if publisher_id_match is None:
+
+            cur.execute("""INSERT INTO publisher (publisher_name)
+                    VALUES (%s)""", (input_game[4],))
+
+            publisher_id_match = get_pub_id(input_game, cur)
+
+        cur.close()
```
Comment (Eibironke) [inline pipeline/load/load.py:116]: Should you close the cur when using with?
```suggestion
      
```

File: pipeline/load/load.py [hunk 4]
```diff
@@ -30,22 +30,113 @@ def get_db_connection(config) -> connection:
...
+            developer_id_match = get_dev_id(input_game, cur)
+
+        cur.close()
+
+    conn.commit()
+
+    return developer_id_match['developer_id']
+
+
+def get_pub_id(input_game: list, cursor: connection.cursor) -> RealDictRow:
```
Comment (shindym) [inline pipeline/load/load.py:88]: Why pass the cursor obj here?

File: pipeline/load/load.py (hunk 2 above)
Comment (Eibironke) [inline pipeline/load/load.py:81]:
```suggestion
       
```

File: pipeline/load/load.py [hunk 5]
```diff
@@ -30,22 +30,113 @@ def get_db_connection(config) -> connection:
...
+
             cur.execute(
-                """INSERT INTO game (name, description, price, developer, publisher, 
-                release_date, rating, website_id)
-                VALUES (%s, %s, %s, %s, %s, %s, %s, %s)""", (game[:-2]))
+                """INSERT INTO game (name, description, price, developer_id, 
+                publisher_id, release_date, rating, website_id)
+                VALUES (%s, %s, %s, %s, %s, %s, %s, %s)""", (game[:8]))
 
         cur.close()
```
Comment (Eibironke) [inline pipeline/load/load.py:141]:
```suggestion
       
```

**Review by Eibironke — CHANGES_REQUESTED** [review]

File: pipeline/load/load.py (hunk 4 above)
Comment (hhm970) [reply pipeline/load/load.py:88]: I appreciate that what I've done by passing the `connection.cursor` object as an argument is quite unorthodox.

I call this function on line 107 in `input_game_pub_get_pub_id()`, when I'm inside a `with conn.cursor() as cur:` expression. 

Do you think I should integrate `get_pub_id()` into this function, or shall I leave it how it is?

File: pipeline/load/load.py (hunk 2 above)
Comment (hhm970) [reply pipeline/load/load.py:81]: Apologies, I thought this was good practice! Will review this!

Comment (hhm970) [reply pipeline/load/load.py]: Thanks! I've implemented these changes now

File: pipeline/load/load.py (hunk 4 above)
Comment (shindym) [reply pipeline/load/load.py:86]: Ah didn't realise. In that case I see why you did that!

**Review by Eibironke — APPROVED** [review]
Looks good 👍

**Pull request merged**